control.lua

```diff
@@ -292,7 +248,7 @@ else
 			--log(serpent.block(c.name))
 			--log(serpent.block(c.position))
 
-			if c.name ~= "iron-rock" and c.name ~= 'seaweed' and c.name ~= 'fish' then
+			if RocksSet[c.name] == nil and c.name ~= 'seaweed' and c.name ~= 'fish' then
```

Comment:
Not deleting generated resources is important because otherwise there are occasions when the next chunk is generated it will delete the resource generated on the nearby chunk leaving map artifacts.